Regression(288453@main) Unexpectedly removed public API on DOMCSSRule

Ahmad-S792 opened the pull request:
#### 95d82b92cca5d3eee24936932862407709e559a3
<pre>
Regression(288453@main) Unexpectedly removed public API on DOMCSSRule
<a href="https://bugs.webkit.org/show_bug.cgi?id=287513">https://bugs.webkit.org/show_bug.cgi?id=287513</a>
<a href="https://rdar.apple.com/144642410">rdar://144642410</a>

Reviewed by Tim Nguyen and Brent Fulgham.

This patch reintroduces what was removed unintentionally in 288453@main,
to ensure that we don&apos;t break public API.

* Source/WebKitLegacy/mac/DOM/DOMCSSRule.h:

Canonical link: <a href="https://commits.webkit.org/290248@main">https://commits.webkit.org/290248@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/75e71d25915524f8a164fa5c215d719c9cb9d00e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89391 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8916 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44231 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94378 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40152 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9304 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17194 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68869 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26538 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92393 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7143 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81132 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49229 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6888 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35517 "Found 2 new test failures: http/tests/security/clipboard/copy-paste-html-cross-origin-iframe-across-origin.html imported/w3c/web-platform-tests/cookies/schemeful-same-site/schemeful-websockets.sub.tentative.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39259 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77245 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36508 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96206 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16572 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12192 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77738 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16827 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76927 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77044 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19017 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21474 "Passed tests") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20068 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9722 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16585 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21896 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16326 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19777 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18107 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKitLegacy/mac/DOM/DOMCSSRule.h:
```diff
@@ -30,6 +30,7 @@
 @class NSString;
 
 enum {
+    DOM_UNKNOWN_RULE = 0,
```
It's questionable that this is even needed, but avoiding breaking external users is worthwhile I suppose -- even for something deprecated in late 2018.

Ahmad-S792 committed:
Regression(288453@main) Unexpectedly removed public API on DOMCSSRule
https://bugs.webkit.org/show_bug.cgi?id=287513
rdar://144642410

Reviewed by Tim Nguyen and Brent Fulgham.

This patch reintroduces what was removed unintentionally in 288453@main,
to ensure that we don't break public API.

* Source/WebKitLegacy/mac/DOM/DOMCSSRule.h:

Canonical link: https://commits.webkit.org/290248@main
